api/v1beta1/cluster_types.go

```diff
@@ -451,6 +469,77 @@ type ClusterStatus struct {
 	// ObservedGeneration is the latest generation observed by the controller.
 	// +optional
 	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+
+	// v1beta2 groups all the fields that will be added or modified in Cluster's status with the V1Beta2 version.
+	// +optional
+	V1Beta2 *ClusterV1Beta2Status `json:"v1beta2,omitempty"`
+}
+
+// ClusterV1Beta2Status groups all the fields that will be added or modified in Cluster with the V1Beta2 version.
+// See https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20240916-improve-status-in-CAPI-resources.md#machineset-newconditions for more context.
+type ClusterV1Beta2Status struct {
+	// conditions represents the observations of a Cluster's current state.
+	// Known condition types are Available, InfrastructureReady, ControlPlaneInitialized, ControlPlaneAvailable, WorkersAvailable, MachinesReady
+	// MachinesUpToDate, RemoteConnectionProbe, ScalingUp, ScalingDown, Remediating, Deleting, Paused.
+	// Additionally, a TopologyReconciled will be added in case the Cluster is referencing a ClusterClass / defining a managed Topology.
+	// +optional
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// controlPlane groups all the observations about Cluster's ControlPlane current state.
+	// +optional
+	ControlPlane *ClusterControlPlaneStatus `json:"controlPlane,omitempty"`
+
+	// workers groups all the observations about Cluster's Workers current state.
+	// +optional
+	Workers *WorkersStatus `json:"workers,omitempty"`
+}
+
+// ClusterControlPlaneStatus groups all the observations about control plane current state.
+type ClusterControlPlaneStatus struct {
+	// desiredReplicas is the total number of desired control plane machines in this cluster.
+	// +optional
+	DesiredReplicas int32 `json:"desiredReplicas"`
+
+	// replicas is the total number of control plane machines in this cluster.
+	// +optional
+	Replicas int32 `json:"replicas"`
+
+	// upToDateReplicas is the number of up-to-date control plane machines in this cluster. A machine is considered up-to-date when Machine's UpToDate condition is true.
+	// +optional
+	UpToDateReplicas int32 `json:"upToDateReplicas"`
+
+	// readyReplicas is the total number of ready control plane machines in this cluster. A machine is considered ready when Machine's Ready condition is true.
+	// +optional
+	ReadyReplicas int32 `json:"readyReplicas"`
+
+	// availableReplicas is the total number of available control plane machines in this cluster. A machine is considered available when Machine's Available condition is true.
+	// +optional
+	AvailableReplicas int32 `json:"availableReplicas"`
+}
+
+// WorkersStatus groups all the observations about workers current state.
+type WorkersStatus struct {
```

Comment:
Same fields as control plane so the feedback there applies here too

Reply:
Fixed